python/ray/dag/compiled_dag_node.py

```diff
@@ -1690,13 +1706,20 @@ async def execute_async(
                 inp = RayDAGArgs(args=args, kwargs=kwargs)
 
             await self._dag_submitter.write(inp)
-            # Allocate a future that the caller can use to get the result.
-            fut = asyncio.Future()
-            await self._fut_queue.put(fut)
-
-        fut = CompiledDAGFuture(self, self._execution_index, fut)
+            # Allocate a future for each channel that the caller
+            # can use to get the result.
+            channel_futs = {}
+            for chan in self._dag_output_fetchers:
+                fut = asyncio.Future()
+                await self._fut_queues[chan].put(fut)
```

Comment:
QQ: can we use asyncio.gather instead and do all put together?

Reply:
No longer applicable as `dag_output_fetchers` is gotten rid of. Per @stephanie-wang's comment, for each execution index, I fetch results from all channels and cache them in `_result_buffer` rather than fetching results separately for each channel with their individual fetcher (which was what `_dag_output_fetchers` represent).